content/chapters/template-chapter/lab/README.md

```diff
@@ -62,1302 +33,25 @@ student@os:~$ file /usr/bin/ls
-
-So up to this point we've seen that one advantage of processes is that they offer better safety than threads.
-Because they use separate virtual address spaces, sibling processes are better isolated than threads.
-Thus, an application that uses processes can be more robust to errors than if it were using threads.
-
-#### Memory Corruption
-
-Because they share the same address space,  threads run the risk of corrupting each other's data.
-Take a look at the code in `support/sum-array-bugs/memory-corruption/python/`.
-The two programs only differ in how they spread their workload.
-One uses threads while the other uses processes.
-
-Run both programs with and without memory corruption.
-Pass any value as a third argument to trigger the corruption.
-
-```
-student@os:~/.../sum-array-bugs/memory-corruption/python$ python3 memory_corruption_processes.py <number_of_processes>  # no memory corruption
-[...]
-
-student@os:~/.../sum-array-bugs/memory-corruption/python$ python3 memory_corruption_processes.py <number_of_processes> 1  # do memory corruption
-[...]
-```
-
-The one using threads will most likely print a negative sum, while the other displays the correct sum.
-This happens because all threads refer the same memory for the array `arr`.
-What happens to the processes is a bit more complicated.
-
-[Later in this lab](#copy-on-write) we will see that initially, the page tables of all processes point to the same physical frames or `arr`.
-When the malicious process tries to corrupt this array by **writing data to it**, the OS duplicates the original frames of `arr` so that the malicious process writes the corrupted values to these new frames, while leaving the original ones untouched.
-This mechanism is called **Copy-on-Write** and is an OS optimisation so that memory is shared between the parent and the child process, until one of them attempts to write to it.
-At this point, this process receives its own separate copies of the previously shared frames.
-
-Note that in order for the processes to share the `sums` dictionary, it is not created as a regular dictionary, but using the `Manager` module.
-This module provides some special data structures that are allocated in **shared memory** so that all processes can access them.
-You can learn more about shared memory and its various implementations [in the Arena section](#shared-memory).
-
-## Usage of Processes and Threads in `apache2`
+### Practice: Baby steps
 
-We'll take a look at how a real-world application - the `apache2` HTTP server - makes use of processes and threads.
-Since the server must be able to handle multiple clients at the same time, it must therefore use some form of concurrency.
-When a new client arrives, the server offloads the work of interacting with that client to another process or thread.
+We recommend that for practice, you add a subsection for the exercises. This will make it so you can delimit the exercises from the lab's text proper.
```

Comment:
```suggestion
We recommend that for practice, you add a subsection for the exercises.
This will make it so you can delimit the exercises from the lab's text.
```